cli/src/main/java/io/github/tcdl/msb/cli/CliMessageHandler.java

```diff
@@ -0,0 +1,69 @@
+package io.github.tcdl.msb.cli;
+
+import io.github.tcdl.msb.adapters.ConsumerAdapter;
+import io.github.tcdl.msb.acknowledge.AcknowledgementHandlerInternal;
+import io.github.tcdl.msb.api.ExchangeType;
+import io.github.tcdl.msb.api.exception.JsonConversionException;
+
+import java.io.IOException;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+/**
+ * This handler dumps messages from the given topics.
+ *
+ * Additionally it examines the content of messages, looks for any other topics mentioned there (for example response topics) and subscribes to them as well.
+ */
+class CliMessageHandler implements ConsumerAdapter.RawMessageHandler {
+    private CliMessageSubscriber subscriber;
+    private boolean prettyOutput;
+    private List<String> follow;
+
+    public CliMessageHandler(CliMessageSubscriber subscriber, List<String> follow, boolean prettyOutput) {
+        this.subscriber = subscriber;
+        this.follow = follow;
+        this.prettyOutput = prettyOutput;
+    }
+
+    /**
+     * @throws JsonConversionException if some problems during parsing JSON
+     */
+    @Override
+    public void onMessage(String jsonMessage, AcknowledgementHandlerInternal handler) {
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Use instance object for mapper. https://theartofdev.com/2014/07/20/jackson-objectmapper-performance-pitfall/